src/main/java/Model/Database/Interaction/I_ControllerUnit.java

```diff
@@ -79,6 +79,58 @@ public static T_ControllerUnit retrieve(Connection conn, PreparedStatement ps, R
         return tc;
     }
 
+    public static ArrayList<T_ControllerUnit> retrieveFilteredAll(Connection conn, PreparedStatement ps, ResultSet rs, int flatId, int centralUnitId) throws SQLException {
+
+        // No Filter is being used
+        if (flatId <= 0 && centralUnitId <= 0) {
+            return retrieveAll(conn, ps, rs);
+        }
+
+        // SQL Definition
+        String usedSql = "SELECT " +
+                "* " +
+                "FROM " + T_ControllerUnit.DBTABLE_NAME + " " +
+                "WHERE ";
+
+
+        // add filter rules
+        boolean flatRule = flatId > 0;
+        boolean centralUnitRule = centralUnitId > 0;
+
+        usedSql = (flatRule ? usedSql + T_ControllerUnit.DBTABLE_NAME + ".FlatID=? " : usedSql);
```

Comment:
vedel by si okomentovat co sa tu robi? ze ak je filter na flatId a na centralUnitId tak sa pridaju do SQL podmienky na to. inak co ak sa nepouziju tie filtre, tam v hornom SQLku zostane prazdny WHERE?

Reply:
Okomentujem no